app/model/Payment/ReadModel/QueryHandlers/RepaymentCandidateListQueryHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Model\Payment\ReadModel\QueryHandlers;
+
+use eGen\MessageBus\QueryBus\IQueryBus;
+use Model\Common\Repositories\IParticipantRepository;
+use Model\DTO\Payment as DTO;
+use Model\Event\SkautisCampId;
+use Model\Payment\Group\Type;
+use Model\Payment\Payment\State;
+use Model\Payment\ReadModel\Queries\PaymentListQuery;
+use Model\Payment\ReadModel\Queries\RepaymentCandidateListQuery;
+use Model\Payment\Repositories\IGroupRepository;
+use function array_column;
+use function array_filter;
+use function array_key_exists;
+use function array_map;
+use function assert;
+
+final class RepaymentCandidateListQueryHandler
+{
+    private IGroupRepository $groups;
+
+    private IQueryBus $queryBus;
+
+    private IParticipantRepository $participants;
+
+    public function __construct(IQueryBus $queryBus, IGroupRepository $groups, IParticipantRepository $participants)
+    {
+        $this->queryBus     = $queryBus;
+        $this->groups       = $groups;
+        $this->participants = $participants;
+    }
+
+    /**
+     * @return DTO\RepaymentCandidate[]
+     */
+    public function __invoke(RepaymentCandidateListQuery $query) : array
+    {
+        $payments   = array_filter(
+            $this->queryBus->handle(new PaymentListQuery($query->getGroupId())),
+            function (DTO\Payment $payment) : bool {
+                return $payment->getState()->equalsValue(State::COMPLETED);
+            }
```

Comment:
Tady lze použít kratší zápis:
```suggestion
            fn (DTO\Payment $payment) => $payment->getState()->equalsValue(State::COMPLETED),
```